Cpp_test/lib/ipc-shm-receive.cpp

```diff
@@ -2,19 +2,29 @@
 
 IPCShmReceive::~IPCShmReceive()
 {
-
-	shm_unlink(this->info.method_name);
-	if (pthread_cond_destroy(&(this->shm_ptr->cond_re)) != 0)
-	{
-		std::cerr << "Error at pthread_cond_destroy(): receive" << std::endl;
-	}
-	if (pthread_cond_destroy(&(this->shm_ptr->cond)) != 0)
-	{
-		std::cerr << "Error at pthread_cond_destroy(): send" << std::endl;
-	}
-	if (pthread_mutex_destroy(&(this->shm_ptr->mutex)) != 0)
+	if (this->shm_ptr != nullptr)
 	{
-		std::cerr << "Error at pthread_mutex_destroy()" << std::endl;
+		int ret1 = pthread_cond_destroy(&(this->shm_ptr->cond));
+		if (ret1 != 0)
+		{
+			std::cerr << "Error at pthread_cond_destroy(): " << strerror(ret1) << std::endl;
+		}
+
+		pthread_mutex_trylock(&(this->shm_ptr->mutex));
```

Comment:
You don't check the return of trylock so you can't be sure that the lock is effective.

Reply:
Because I can not use unlock_mutex() in destructor so I do not need to use trylock here. I will remove it.

Reply:
I mean the method you created `unlock_mutex()` can throw, so you shall not called it in the destructor (but you can call pthread_mutex_unlock() )